### PR TITLE
Default to False output for null streamlines in streamline_near_roi

### PR DESCRIPTION
**Pull request by DanNBullock**
Per [the discussion on the issue thread](https://github.com/dipy/dipy/issues/2509), I went ahead and cleaned up [the test_utils section where I had created a test for null streamlines](https://github.com/DanNBullock/dipy/blob/01df7ba4042c90f1d7dc3eedac55bb598c21bb31/dipy/tracking/tests/test_utils.py#L357-L371).  It's slightly inelegant, in that I have to append a non-null streamline to the front in order to bypass issues with near_roi and tol/dtc computations, but it works as implemented.

The [streamline_near_roi change itself](https://github.com/DanNBullock/dipy/blob/01df7ba4042c90f1d7dc3eedac55bb598c21bb31/dipy/tracking/utils.py#L706-L707) appears to be effective and well behaved.

Let me know if additional adjustments are suggested/required.

### CODE REVIEW

File: dipy/tracking/tests/test_utils.py [hunk 1]
```diff
@@ -353,11 +354,27 @@ def test_near_roi():
...
+                           np.array([True, False, False, False]))
+    npt.assert_array_equal(near_roi(streamlinesNULL, np.eye(4), mask),
+                           np.array([False, False, False, False]))
+
     # If there is an affine, we need to use it:
     affine = np.eye(4)
     affine[:, 3] = [-1, 100, -20, 1]
     # Transform the streamlines:
-    x_streamlines = [sl + affine[:3, 3] for sl in streamlines]
+    x_streamlines = [sl + affine[:3, 3] for sl in streamlines[-3:]]
```
Comment (arokem) [inline dipy/tracking/tests/test_utils.py:377]: Why is this needed? It doesn't seem to be referring to the new variable that you created. Is that what you meant to do here? I.e., 

```
 x_streamlines = [sl + affine[:3, 3] for sl in streamlinesNull[-3:]]
```

File: dipy/tracking/tests/test_utils.py [hunk 2]
```diff
@@ -353,11 +354,27 @@ def test_near_roi():
     npt.assert_array_equal(near_roi(streamlines, np.eye(4), mask),
                            np.array([False, True, False]))
 
+    # test for handling of various forms of null streamlines
+    # including a streamline from previous test because near_roi / tol
+    # can't handle completely empty streamline collections
+    streamlinesNULL = [np.array([[0., 0., 0.9],
```
Comment (arokem) [inline dipy/tracking/tests/test_utils.py:360]: PEP8:

```suggestion
    streamlines_null = [np.array([[0., 0., 0.9],
```

**Review by arokem — COMMENTED** [review]
Thanks for adding this! I had a couple of small questions/suggestions

File: dipy/tracking/tests/test_utils.py (hunk 1 above)
Comment (DanNBullock) [reply dipy/tracking/tests/test_utils.py:377]: I actually don't know what this is... this (to my understanding) doesn't really have anything to do with the changes I'd intended.  It might be a typo?  I believe it should be left in the previous state (i.e. no [-3:])

File: dipy/tracking/tests/test_utils.py (hunk 2 above)
Comment (DanNBullock) [reply dipy/tracking/tests/test_utils.py]: Fixed [here]( https://github.com/dipy/dipy/pull/2515/commits/65664c88663003f1b6f25b2c5d22c5a83f6cdbd3) I think.

However, [it seems the pep8 styling is off](https://github.com/dipy/dipy/pull/2515#issuecomment-1038463545)?  In my spyder IDE it didn't seem to have a problem with this, so I'm not entirely sure what the specific fix is.

Comment (arokem) [reply dipy/tracking/tests/test_utils.py]: From [PEP8](https://www.python.org/dev/peps/pep-0008/#naming-conventions): "mixedCase is allowed only in contexts where that's already the prevailing style (e.g. threading.py), to retain backwards compatibility."

Comment (arokem) [reply dipy/tracking/tests/test_utils.py]: FWIW, I agree with the automated pep8 that those lines are under-indented. One more space :-)

Comment (DanNBullock) [reply dipy/tracking/tests/test_utils.py]: I could have sworn I fixed some of this in an earlier push.  In any case [I believe it is fixed now](https://github.com/dipy/dipy/pull/2515/commits/5a59df375109406b4131f9c9d1b692a377b53eab).  Apologies for the false starts on this.  I suppose that even after all these years of working with git I'm still a bit unpracticed.

**Comment by arokem** [comment]
LGTM! Any other comments from anyone? I can go ahead and merge this tomorrow if not.

**Review by skoudoro — APPROVED** [review]
+1 for merging.

you can go ahead @arokem

Thanks @DanNBullock !
